src/_controller/authentification/auth.service.ts

```diff
@@ -4,28 +4,41 @@ import { User } from '../_database/_entity/user/user.entity'
 import { ConfigService } from '@nestjs/config'
 import * as bcrypt from 'bcrypt'
 import { MailService } from 'src/_helper/mail/mail.service'
+import { JwtService } from '@nestjs/jwt';
 
 @Injectable()
 export class AuthService {
     constructor(
         private readonly _userService: UserService,
+        private readonly _jwtService: JwtService,
         private readonly _configService: ConfigService,
         private readonly _mailService: MailService
     ) { }
 
-    async signIn(login: string, password: string): Promise<any> {
-        const userLogin = await this._userService.findLogin(login)
-        console.log(userLogin)
+    async validateUser(login: string, password: string): Promise<any> {
+        if (login.trim() === '' || password.trim() === '') {
+            throw new UnauthorizedException()
+        }
+
+        const userLogin = await await this._userService.findLogin(login)
         const isPasswordMatching = await bcrypt.compare(
             password,
             userLogin.password
         )
+        
         if (!userLogin || !isPasswordMatching) {
             throw new UnauthorizedException()
         }
-        return 'Connecté'
+
+        return userLogin
     }
 
+    async signIn(user: User) {
+        return {
+            token: this._jwtService.sign({ user: user }),
+            user: user
+        }
+    }
 
     async signUp(user: User): Promise<any> {
         if (user.password === '') {
```

Comment:
```suggestion
        if (user.password.trim().length === 0) {
```

Pour faire pareil que mon autre suggestion

Reply:
Je peux pas suggérer en dessous, mais en général on fait pas de `else` quand ton `if` `throw` qqch, vu que ça stop l'execution de ta méthode